src/components/mainLayout/signUpForm/signUpForm.tsx

```diff
@@ -41,8 +44,25 @@ const SignUpForm = () => {
 		router.back();
 	};
 
+	const isAxiosError = (error: unknown): error is AxiosError => {
+		return (error as AxiosError).isAxiosError !== undefined;
+	};
+
 	const onSubmit = async (data: ISignUpForm) => {
-		console.log(data);
+		try {
+			await registration(data);
+			router.push(MainPath.Login);
+		} catch (error) {
+			if (
+				isAxiosError(error) &&
+				error.response &&
+				error.response.status &&
+				error.response.status >= ApiResponseCode.ERROR_STATUS_MIN &&
+				error.response.status < ApiResponseCode.ERROR_STATUS_MAX
+			) {
+				router.push(MainPath.ServerError);
+			}
+		}
```

Comment:
router.push(MainPath.ServerError);

Function or block with a single if block is bad practice. This case worsens code readability. In this case, it is better to leave a mock operation